perf/perftool.py

```diff
@@ -57,10 +66,21 @@ def test(self):
             if '-- [ FAIL ] --' in line:
                 self.count += 1
                 self.log.info(line)
-
         if self.count > 0:
             self.fail("%s Test failed" % self.count)
 
+        process.system_output("perf test")
+        tarball = self.fetch_asset('https://github.com/deater/'
+                                   'perf_event_tests/archive/'
+                                   'master.zip', expire='7d')
+        archive.extract(tarball, self.srcdir)
+        self.srcdir = os.path.join(self.srcdir, 'perf_event_tests-master')
+        build.make(self.srcdir)
+        os.chdir(self.srcdir)
+        process.system_output("echo -1 >/proc/sys/kernel/"
+                              "perf_event_paranoid")
+        process.system_output("./run_tests.sh")
+
```

Comment:
You are running 2 tests here: perftool and perf_event_tests.
Make sure the failure/error are indicated better for each of them seperately, instead of generic "Test Failed".
I would suggest having seperate `test_` functions for each of them.